### DISCUSSION TITLE
Restrict user permissions (no editing)

### DISCUSSION THREAD

**Discussion by blumarten** [mealie-recipes/mealie#2475] [Feature Request]
please may we have a way of restricting users from edting recipes? either by having a read-only user class, or being able to toggle editing as an indivdual user permission (with a default of OFF).. or, alternatively, a way of mass locking recipes.. (the current feature is there, but disabled)

thanks for any consideration!

**Comment by tshirtnjeans** [comment]
In the old version I could share the link with anyone and they could search and view any recipe, but it was read only. You had to sign into admin to add, change or remove anything. Now with v1, the link sends you to a login page, and you need to have an acct to see any recipes. Even if you create a "guest" account, there's no read-only option. Any account with access to a group's recipes can modify them.

I'd prefer if the main page was all the public recipes from any group and there be a login button somewhere on there, i.e. it defaults to visitors as guests but sign-in is required to see private group recipes and modify anything. 

Now with v1, you need to login to see any recipe. If you have access to a recipe, you have full read/write/remove access. I don't want anyone changing or removing recipes but the owner/creator. You can only share single recipes at a time via a link and that link HAS to have an expiry date. Links shouldn't expire by default, but have the option to. 

Also, the copy to clipboard button hasn't been working for me, I wish the actual content it's trying to copy was visible.

**Reply by djmega152** [reply]
I'd love to have a read-only user option.  I want to be able to share the recipe database with friends but I don't want them adding new recipes or modifying/deleting existing recipes.

**Reply by michael-genson** [reply]
FYI this has been supported in v1 for quite some time now:

https://docs.mealie.io/documentation/getting-started/usage/permissions-and-public-access/#public-recipe-access

**Comment by swanemar** [comment]
Just stumbled upon Mealie and this is a definite wish from me as well:)

**Reply by michael-genson** [reply]
This is an older thread; this has been available in v1 for quite some time now:

https://docs.mealie.io/documentation/getting-started/usage/permissions-and-public-access/#public-recipe-access

Assuming proper settings, you can visit your group page without logging in (which is effectively a "read-only" user)
